File: apps/chat-ios/AmazonChimeSDKMessagingDemo/messaging-websocket/session/MessagingSessionObserver.swift
//
//  MessagingSessionObserver.swift
//  AmazonChimeSDKMessagingDemo
//
//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
//  SPDX-License-Identifier: Apache-2.0
//

import Foundation

@objc public protocol MessagingSessionObserver {

    func onMessagingSessionStarted()

    func onMessagingSessionConnecting(reconnecting: Bool)

    func onMessagingSessionStopped(status: MessagingSessionStatus)

    func onMessagingSessionMessageReceived(message: Message)
}

File: apps/chat-android/messaging-websocket/src/main/java/com/amazonaws/services/chime/sdk/messaging/session/MessagingSessionObserver.kt
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

package com.amazonaws.services.chime.sdk.messaging.session

import com.amazonaws.services.chime.sdk.messaging.Message

interface MessagingSessionObserver {
    fun onMessagingSessionStarted()

    fun onMessagingSessionConnecting(reconnecting: Boolean)

    fun onMessagingSessionStopped(status: MessagingSessionStatus)

    fun onMessagingSessionReceivedMessage(message: Message)
}
